src/main/resources/assets/opencomputers/loot/openos/lib/uuid.lua

```diff
@@ -5,13 +5,21 @@ function uuid.next()
   -- 8-4-4-4-12 (halved sizes because bytes make hex pairs)
   local sets = {4, 2, 2, 2, 6}
   local result = ""
+  local pos = 0
 
   for _,set in ipairs(sets) do
     if result:len() > 0 then
       result = result .. "-"
     end
     for i = 1,set do
-      result = result .. string.format("%02x", math.random(0, 255))
+      local byte = math.random(0, 255)
+      if pos == 6 then
+        byte = bit32.bor(bit32.band(byte, 0x0F), 0x40)
+      elseif pos == 8 then
+        byte = bit32.bor(bit32.band(byte, 0x3F), 0x80)
+      end
+      result = result .. string.format("%02x", byte)
+      pos = pos + 1
```

Comment:
Can one just use `i` instead of maintaining a separate counter?

Reply:
The value of i is from 1 to the value of 4, 1 or 8 depending on the part of the uuid being generated. Thus you would still need to maintain an index of which section of the uuid you're manipulating.

For example the 2 options are the method I used, or a method where the option for byte 6 is byte 1 or part 3 and byte 8 is byte 1 of part 4. So you could change "for _,set in ipairs(sets) do" to "for part,set in ipairs(sets) do" and then use "if part == 3 and i == 1 then" and "if part == 4 and i == 1" instead of "if pos == 6 then" and "if pos == 8 then" respectively

Reply:
Ahhh, woops. I evidently failed to read the code properly - sorry.